packages/app/src/systems/Send/machines/sendMachine.ts

```diff
@@ -0,0 +1,298 @@
+          onDone: [
+            unlockMachineErrorAction('unlocking', 'unlockError'),
+            { target: 'idle', actions: 'assignWallet' },
+          ],
+        },
+        on: {
+          UNLOCK_WALLET: {
+            actions: [
+              send<MachineContext, UnlockMachineEvents>(
+                (_, ev) => ({ type: 'UNLOCK_WALLET', input: ev.input }),
+                { to: 'unlock' }
+              ),
+            ],
+          },
+        },
+      },
+      idle: {
+        initial: 'checking',
+        on: {
+          BACK: {
+            actions: ['goToHome'],
+          },
+        },
+        states: {
+          checking: {
+            always: [{ target: 'delay', cond: 'canCalcFee' }],
+          },
+          delay: {
+            after: {
+              600: 'creatingTx',
+            },
+          },
+          creatingTx: {
+            invoke: {
+              src: 'createTx',
+              data: (ctx: MachineContext) => ({ input: ctx.inputs }),
+              onDone: [
+                {
+                  target: 'failed',
+                  actions: ['assignTxRequestError'],
+                  cond: FetchMachine.hasError,
+                },
+                {
+                  actions: ['assignTxRequest'],
+                  target: 'waitingConfirm',
+                },
+              ],
+            },
+          },
+          waitingConfirm: {
+            on: {
+              CONFIRM: {
+                target: '#(machine).confirming',
+              },
+            },
+          },
+          failed: {
+            type: 'final',
+          },
+        },
+      },
+      confirming: {
+        initial: 'idle',
+        on: {
+          BACK: {
+            target: 'idle.waitingConfirm',
+          },
+        },
+        states: {
+          idle: {
+            on: {
+              CONFIRM: {
+                target: 'sendingTx',
+              },
+            },
+          },
+          sendingTx: {
+            invoke: {
+              src: 'sendTx',
+              data: (ctx: MachineContext) => ({ input: ctx }),
+              onDone: [
+                {
+                  target: 'idle',
+                  actions: ['assignTxApproveError'],
+                  cond: FetchMachine.hasError,
+                },
+                {
+                  actions: ['assignTxApprove'],
+                  target: 'success',
+                },
+              ],
+            },
+          },
+          success: {
+            type: 'final',
+            entry: ['showSuccessMessage', 'goToHome'],
+          },
+        },
+      },
+    },
+    on: {
+      SET_ASSET: {
+        actions: ['assignAsset'],
+        target: 'idle',
+      },
+      SET_ADDRESS: {
+        actions: ['assignAddress'],
+        target: 'idle',
+      },
+      SET_AMOUNT: {
+        actions: ['assignAmount'],
+        target: 'idle',
+      },
+      RESET: {
+        actions: ['reset'],
+        target: 'idle',
+      },
+    },
```

Comment:
all this events are global in the machine?

looks like they are specific of `idle` state? if yes let's move them to right state

Reply:
Nope, they need to be global, because if the machine are in the `checking` state getting info about transaction, and the user start to change some of the form values, the machine need to go back to the `idle`

Reply:
interesting